File: sdk/javascript/packages/hjtpx-sdk/examples/basic-examples.ts
import { CaptchaClient, CaptchaType } from 'hjtpx-sdk';

async function exampleImageCaptcha() {
  console.log('\n=== Image Captcha Example ===');

  const client = new CaptchaClient({
    baseUrl: 'http://localhost:8080',
    debugMode: true,
  });

  const captcha = await client.generateImageCaptcha({
    type: CaptchaType.MIXED,
    count: 4,
    noiseMode: 2,
    lineMode: 1,
  });

  console.log('✓ Challenge ID:', captcha.challenge_id);

  const result = await client.verifyImageCaptcha(captcha.challenge_id, '1234');
  console.log('✓ Verification success:', result.success);

  const stats = client.getStats();
  console.log('\n📊 Statistics:');
  console.log('  Total Requests:', stats.totalRequests);
  console.log('  Success Rate:', stats.successRate.toFixed(2) + '%');
}

async function exampleSliderCaptcha() {
  console.log('\n=== Slider Captcha Example ===');

  const client = new CaptchaClient({
    baseUrl: 'http://localhost:8080',
  });

  const slider = await client.generateSliderCaptcha({
    width: 360,
    height: 220,
  });

  console.log('✓ Challenge ID:', slider.challenge_id);
  console.log('  Slider Size:', slider.slider_width + 'x' + slider.slider_height);

  const result = await client.verifySliderCaptcha(slider.challenge_id, '120');
  console.log('✓ Verification success:', result.success);
  console.log('  Score:', result.score);
  console.log('  Risk Level:', result.risk_level);
}

async function exampleClickCaptcha() {
  console.log('\n=== Click Captcha Example ===');

  const client = new CaptchaClient({
    baseUrl: 'http://localhost:8080',
  });

  const click = await client.generateClickCaptcha({
    width: 360,
    height: 220,
    iconCount: 4,
  });

  console.log('✓ Challenge ID:', click.challenge_id);
  console.log('  Target Index:', click.target_index);
  console.log('  Icon Positions:', click.icon_positions);

  const clicks = [
    {
      x: click.target_position[0],
      y: click.target_position[1],
      duration: 500,
    },
  ];

  const result = await client.verifyClickCaptcha(click.challenge_id, clicks);
  console.log('✓ Verification success:', result.success);
}

async function exampleErrorHandling() {
  console.log('\n=== Error Handling Example ===');

  const client = new CaptchaClient({
    baseUrl: 'http://localhost:8080',
    maxRetries: 0,
  });

  try {
    await client.verifyImageCaptcha('', '1234');
  } catch (error) {
    console.log('✓ Caught expected error:', error.message);
    console.log('  Error Code:', error.code);
    console.log('  Is Invalid Params:', error.isInvalidParams());
  }

  try {
    await client.verifySliderCaptcha('test-id', '');
  } catch (error) {
    console.log('✓ Caught expected error:', error.message);
  }
}

async function exampleStatistics() {
  console.log('\n=== Statistics Example ===');

  const client = new CaptchaClient({
    baseUrl: 'http://localhost:8080',
  });

  await client.generateImageCaptcha();
  await client.generateSliderCaptcha();
  await client.generateClickCaptcha();

  const stats = client.getStats();
  console.log('📊 Current Statistics:');
  console.log('  Total Requests:', stats.totalRequests);
  console.log('  Successful:', stats.successfulRequests);
  console.log('  Failed:', stats.failedRequests);
  console.log('  Retried:', stats.retriedRequests);
  console.log('  Success Rate:', stats.successRate.toFixed(2) + '%');
}

async function main() {
  console.log('='.repeat(50));
  console.log('hjtpx JavaScript/TypeScript SDK Examples');
  console.log('='.repeat(50));

  try {
    await exampleImageCaptcha();
    await exampleSliderCaptcha();
    await exampleClickCaptcha();
    await exampleErrorHandling();
    await exampleStatistics();

    console.log('\n' + '='.repeat(50));
    console.log('All examples completed successfully!');
    console.log('='.repeat(50));
  } catch (error) {
    console.error('\n✗ Error:', error);
  }
}

main();

File: sdk/python/examples/basic_examples.py
"""
hjtpx Python SDK - Examples
"""

import time
from hjtpx import CaptchaClient, Config
from hjtpx import (
    ImageCaptchaRequest,
    SliderCaptchaRequest,
    ClickCaptchaRequest,
    ClickData,
    CaptchaType,
)


def example_image_captcha():
    """Image Captcha Example"""
    print("\n=== Image Captcha Example ===")

    client = CaptchaClient()
    client.set_debug_mode(True)

    request = ImageCaptchaRequest(
        captcha_type=CaptchaType.MIXED,
        count=4,
        noise_mode=2,
        line_mode=1,
    )

    captcha = client.generate_image_captcha(request)
    print(f"✓ Challenge ID: {captcha.challenge_id}")

    result = client.verify_image_captcha(captcha.challenge_id, "1234")
    print(f"✓ Verification success: {result.success}")

    stats = client.get_stats()
    print(f"\n📊 Statistics:")
    print(f"  Total Requests: {stats.total_requests}")
    print(f"  Success Rate: {stats.success_rate:.2f}%")


def example_slider_captcha():
    """Slider Captcha Example"""
    print("\n=== Slider Captcha Example ===")

    client = CaptchaClient()

    request = SliderCaptchaRequest(width=360, height=220)
    slider = client.generate_slider_captcha(request)
    print(f"✓ Challenge ID: {slider.challenge_id}")
    print(f"  Slider Size: {slider.slider_width}x{slider.slider_height}")

    result = client.verify_slider_captcha(slider.challenge_id, "120")
    print(f"✓ Verification success: {result.success}")
    print(f"  Score: {result.score}")
    print(f"  Risk Level: {result.risk_level}")

    stats = client.get_stats()
    print(f"\n📊 Statistics:")
    print(f"  Total Requests: {stats.total_requests}")
    print(f"  Success Rate: {stats.success_rate:.2f}%")


def example_click_captcha():
    """Click Captcha Example"""
    print("\n=== Click Captcha Example ===")

    client = CaptchaClient()

    request = ClickCaptchaRequest(width=360, height=220, icon_count=4)
    click = client.generate_click_captcha(request)
    print(f"✓ Challenge ID: {click.challenge_id}")
    print(f"  Target Index: {click.target_index}")
    print(f"  Icon Positions: {click.icon_positions}")

    clicks = [
        ClickData(
            x=click.target_position[0],
            y=click.target_position[1],
            duration=500,
        )
    ]

    result = client.verify_click_captcha(click.challenge_id, clicks)
    print(f"✓ Verification success: {result.success}")
    print(f"  Score: {result.score}")

    stats = client.get_stats()
    print(f"\n📊 Statistics:")
    print(f"  Total Requests: {stats.total_requests}")
    print(f"  Success Rate: {stats.success_rate:.2f}%")


def main():
    print("=" * 50)
    print("hjtpx Python SDK Examples")
    print("=" * 50)

    try:
        example_image_captcha()
        example_slider_captcha()
        example_click_captcha()

        print("\n" + "=" * 50)
        print("All examples completed successfully!")
        print("=" * 50)

    except Exception as e:
        print(f"\n✗ Error: {e}")
        raise


if __name__ == "__main__":
    main()
